src/completions/Tab.ts

```diff
@@ -24,17 +24,33 @@ class BufferCompletionOption
         this.tabIndex = tab.index
         this.tabId = tab.id
 
-        // Two character tab properties prefix
-        let pre = ""
-        if (tab.active) pre += "%"
+        // pre contains max four uppercase characters for tab status.
+        // If statusstylepretty is set to true replace use unicode characters,
+        // but keep plain letters in hidden column for completion.
+        let preplain = ""
+        if (tab.active) preplain += "%"
         else if (isAlternative) {
-            pre += "#"
+            preplain += "#"
             this.value = "#"
         }
-        if (tab.pinned) pre += "@"
+        let pre = preplain
+        if (tab.pinned) preplain += "P"
+        if (tab.audible) preplain += "A"
+        if (tab.mutedInfo.muted) preplain += "M"
+        if (tab.discarded) preplain += "D"
+
+        if (config.get("completions", "Tab", "statusstylepretty") === "true") {
+            if (tab.pinned) pre += "\uD83D\uDCCC"
+            if (tab.audible) pre += "\uD83D\uDD0A"
+            if (tab.mutedInfo.muted) pre += "\uD83D\uDD07"
+            if (tab.discarded) pre += "\u2296"
+        } else {
+            pre = preplain
+        }
 
         // Push prefix before padding so we don't match on whitespace
         this.fuseKeys.push(pre)
+        this.fuseKeys.push(preplain)
```

Comment:
Hmm, if we do this it won't do the case sensitive matching.

I was imagining that we would do something more like the matching we do for numbers.

Reply:
Yes, agreed. I don't fully understand how the completion code works yet. Testing out some combinations of tabs does not always give me the selection I expected.